Code/Engine/Foundation/Profiling/Profiling_Tracy.h

```diff
@@ -0,0 +1,98 @@
+#include <Foundation/Basics.h>
+#include <Foundation/Strings/HashedString.h>
+#if EZ_ENABLED(EZ_USE_PROFILING) || defined(EZ_DOCS)
+#  if defined(TRACY_ENABLE)
+#    include <tracy/tracy/Tracy.hpp>
+
+EZ_ALWAYS_INLINE ezUInt32 ___tracyGetStringLength(const char* szString)
+{
+  return ezStringUtils::GetStringElementCount(szString);
+}
+
+EZ_ALWAYS_INLINE ezUInt32 ___tracyGetStringLength(ezStringView szString)
+{
+  return szString.GetElementCount();
+}
+
+EZ_ALWAYS_INLINE ezUInt32 ___tracyGetStringLength(const ezString& szString)
+{
+  return szString.GetElementCount();
+}
+EZ_ALWAYS_INLINE ezUInt32 ___tracyGetStringLength(const ezStringBuilder& szString)
+{
+  return szString.GetElementCount();
+}
+EZ_ALWAYS_INLINE ezUInt32 ___tracyGetStringLength(const ezHashedString& szString)
+{
+  return szString.GetView().GetElementCount();
+}
+EZ_ALWAYS_INLINE const char* __convertezStringToConstChar(const ezString& szString)
+{
+  return szString.GetData();
+}
+EZ_ALWAYS_INLINE const char* __convertezStringToConstChar(const ezStringBuilder& szString)
+{
+  return szString.GetData();
+}
+EZ_ALWAYS_INLINE const char* __convertezStringToConstChar(const ezHashedString& szString)
+{
+  return szString.GetData();
+}
+EZ_ALWAYS_INLINE const char* __convertezStringToConstChar(const ezStringView& szString)
+{
+  ezStringBuilder temp_str;
```

Comment:
This is a typical use after free, you can't return the pointer to the stringbuilder data, when that stringbuilder is local to the function.

Is there no way to give tracy a string view?
Alternatively you would need something else to copy the strings to for a longer time.

@C-Core do you have any idea? Maybe use the frame allocator?

Reply:
Just saw how this was used and fixed it, because Tracy already takes strings as startptr+size.